src/arrayMethodJoin.js

```diff
@@ -4,8 +4,42 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this.length === 1) {
+      return '' + this[0];
+    }
+
+    if (this[1] === undefined) {
+      return this;
+    }
```

Comment:
can you try to do it without these check aswell? Just delete them and play with the conditions below